registry/client_test.go

```diff
@@ -11,76 +12,95 @@ import (
+	{image: "katsubushi/katsubushi", tagOrDigest: "v1.6.0", arch: "amd64", os: "linux"},
+	{image: "katsubushi/katsubushi", tagOrDigest: "sha256:9426deec3ed0c1f41f070509c4e9729ac32d65920b919f0adc09809812c0f247", arch: "amd64", os: "linux"},
+	{image: "public.ecr.aws/mackerel/mackerel-container-agent", tagOrDigest: "plugins", arch: "amd64", os: "linux"},
+	{image: "public.ecr.aws/mackerel/mackerel-container-agent", tagOrDigest: "sha256:c7d960d3f87cc9f9b6e5afee8ad2fe5d0af14ec70d5c089ec1a77304ccb40e2d", arch: "amd64", os: "linux"},
+	{image: "gcr.io/kaniko-project/executor", tagOrDigest: "v0.10.0", arch: "amd64", os: "linux"},
+	{image: "gcr.io/kaniko-project/executor", tagOrDigest: "sha256:78d44ec4e9cb5545d7f85c1924695c89503ded86a59f92c7ae658afa3cff5400", arch: "amd64", os: "linux"},
+	{image: "ghcr.io/github/super-linter", tagOrDigest: "v3", arch: "amd64", os: "linux"},
+	{image: "ghcr.io/github/super-linter", tagOrDigest: "sha256:b1a1417c30483783987536fd8beb64d85f2509d27b5fb72afbe327a22f0ed59b", arch: "amd64", os: "linux"},
+	{image: "mcr.microsoft.com/windows/servercore/iis", tagOrDigest: "windowsservercore-ltsc2019", arch: "amd64", os: "windows"},
+	{image: "mcr.microsoft.com/windows/servercore/iis", tagOrDigest: "sha256:c077419bb006eb5c9a0fe3e5a9cdf018cdcc2b50c5ea11ad39ffbecf93fd3f07", arch: "amd64", os: "windows"},
 }
 
 var testFailImages = []struct {
-	image string
-	tag   string
-	arch  string
-	os    string
+	image       string
+	tagOrDigest string
+	arch        string
+	os          string
 }{
-	{image: "debian", tag: "xxx", arch: "arm64", os: "linux"},
-	{image: "katsubushi/katsubushi", tag: "xxx", arch: "amd64", os: "linux"},
-	{image: "public.ecr.aws/mackerel/mackerel-container-agent", tag: "xxx", arch: "amd64", os: "linux"},
-	{image: "gcr.io/kaniko-project/executor", tag: "xxx", arch: "amd64", os: "linux"},
-	{image: "ghcr.io/github/super-linter", tag: "xxx", arch: "amd64", os: "linux"},
-	{image: "mcr.microsoft.com/windows/servercore/iis", tag: "xxx", arch: "amd64", os: "windows"},
-	{image: "xxx", tag: "xxx", arch: "xxx", os: "xxx"},
+	{image: "debian", tagOrDigest: "xxx", arch: "arm64", os: "linux"},
+	{image: "debian", tagOrDigest: "sha256:xxx", arch: "arm64", os: "linux"},
+	{image: "katsubushi/katsubushi", tagOrDigest: "xxx", arch: "amd64", os: "linux"},
+	{image: "public.ecr.aws/mackerel/mackerel-container-agent", tagOrDigest: "xxx", arch: "amd64", os: "linux"},
+	{image: "gcr.io/kaniko-project/executor", tagOrDigest: "xxx", arch: "amd64", os: "linux"},
+	{image: "ghcr.io/github/super-linter", tagOrDigest: "xxx", arch: "amd64", os: "linux"},
+	{image: "mcr.microsoft.com/windows/servercore/iis", tagOrDigest: "xxx", arch: "amd64", os: "windows"},
+	{image: "xxx", tagOrDigest: "xxx", arch: "xxx", os: "xxx"},
 }
 
 func TestImages(t *testing.T) {
 	for _, c := range testImages {
-		t.Logf("testing %s:%s", c.image, c.tag)
+		var fullImage string
```

Comment:
I thought it would be easier to understand if the tests were written inline, so I haven't created helper functions. If you have any opinions, I will revise it.